data-gen/src/main/java/io/smartcat/data/loader/rules/RangeRuleDouble.java

```diff
@@ -17,8 +18,15 @@
     // definition of the range: e.g [a,b,c,d] : a < b <= c < d is a set of ranges: {[a,b),[c,d)}
     private List<Double> ranges = new ArrayList<>();
 
+    private final List<Double> rangeEdges = new LinkedList<>();
+
     private Randomizer random;
 
+    /**
+     * Small value that will be subtracted from the end of ranges.
+     */
+    public static final Double EPSILON = 0.00000000001;
```

Comment:
If EPSILON is added one more zero before 1 (0.000000000001 instead of 0.00000000001), RangeRuleDoubleTest fails. Not sure why, yet.